independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java

```diff
@@ -533,7 +533,18 @@ private DefaultRepositorySystemSession newRepositorySystemSession() throws Boots
         DefaultSettingsDecryptionRequest decrypt = new DefaultSettingsDecryptionRequest();
         decrypt.setProxies(settings.getProxies());
         decrypt.setServers(settings.getServers());
+        // need to set `settings-security.xml` location extra, because it isn't discovered
+        // by BeanBag when constructing `DefaultSecDispatcher`
```

Comment:
Should it be? Or this is the experted behavior?

Wondering as it would be nice not having to think about that every time.

Reply:
This code existed before my previous PR, where I tried to make the code a lot closer to the Maven original. The comment explains why it was there and why it should be there again.

I'm honestly not sure whether BeanBag actually _could_ discover the location itself, it's declared as

```java
@Inject
public DefaultSecDispatcher( final PlexusCipher _cipher,
                             final Map<String, PasswordDecryptor> _decryptors,
                             @Named( "${_configurationFile:-" + DEFAULT_CONFIGURATION + "}" )
                             final String _configurationFile )
{
    ...
}
```

It seems to me that this is out of BeanBag's reach.

Reply:
Yeah, it's not the best but that's what we had before, e.g. https://github.com/quarkusio/quarkus/blob/3.15/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java#L541-L549